src/k8s/cmd/k8s/k8s_x_wait_for_dns.go

```diff
@@ -0,0 +1,20 @@
+package k8s
+
+import (
+	cmdutil "github.com/canonical/k8s/cmd/util"
+	"github.com/canonical/k8s/pkg/k8sd/features"
+	"github.com/spf13/cobra"
+)
+
+func newXWaitForDNSCmd(env cmdutil.ExecutionEnvironment) *cobra.Command {
+	waitForDNSCmd := &cobra.Command{
+		Use:    "x-wait-for-dns",
+		Short:  "Wait for DNS to be ready",
+		Hidden: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			features.StatusChecks.CheckDNS(cmd.Context(), env.Snap)
```

Comment:
then, this command should have a timeout and use `control.WaitUntilReady`, calling CheckDNS repeatedly